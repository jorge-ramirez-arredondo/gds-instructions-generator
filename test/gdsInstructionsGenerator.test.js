const gdsInstructionsGenerator = require("../src/gdsInstructionsGenerator");
const sampleAST = require("./sampleAST");

describe('gdsInstructionsGenerator', function() {
  it("generates expected instructions", function () {
    expect(gdsInstructionsGenerator(sampleAST)).toMatchSnapshot();
  });
});
