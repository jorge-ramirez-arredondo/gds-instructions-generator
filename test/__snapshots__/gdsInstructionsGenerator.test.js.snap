// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gdsInstructionsGenerator generates expected instructions 1`] = `
"{
		type: \\"Script\\",
		instructions: [
			{
		type: \\"Dialog\\",
		execute: (_scriptScope) => {
			_scriptScope.dialog += \\"\\\\r\\\\n\\\\r\\\\n\\";
		}
	},
{
		type: \\"Dialog\\",
		execute: (_scriptScope) => {
			_scriptScope.dialog += \\"\\\\r\\\\n\\\\r\\\\nHere is the story text.\\\\r\\\\n\\\\r\\\\nDown bellow is js code:\\\\r\\\\n\\";
		}
	},
{
		type: \\"JSExpression\\",
		execute: (_scriptScope) => {
			const test = {
    test: \\"{{}}\\"
  };
  const test2 = \\"}}\\";
  const test3 = \`\${test ? ({}) : '{\\"}}\\"}'}\`;
		}
	},
{
		type: \\"Dialog\\",
		execute: (_scriptScope) => {
			_scriptScope.dialog += \\"\\\\r\\\\n\\\\r\\\\n123\\";
		}
	},
{
		type: \\"JSExpression\\",
		execute: (_scriptScope) => {
			#asd \\"test\\"
		}
	},
{
		type: \\"Dialog\\",
		execute: (_scriptScope) => {
			_scriptScope.dialog += \\"\\\\r\\\\n1231\\\\r\\\\n\\";
		}
	}
		]
	}"
`;
